src/main/java/com/zendesk/maxwell/schema/MysqlSavedSchema.java

```diff
@@ -226,19 +226,30 @@ private void executeColumnInsert(Connection conn, ArrayList<Object> columnData)
 		columnData.clear();
 	}
 
-	public static MysqlSavedSchema restore(Connection connection, MaxwellContext context) throws SQLException, IOException, InvalidSchemaError {
-		MysqlSavedSchema s = new MysqlSavedSchema(context.getServerID(), context.getCaseSensitivity());
+	public static MysqlSavedSchema restore(MaxwellContext context, BinlogPosition targetPosition) throws SQLException, InvalidSchemaError {
+		try ( Connection conn = context.getMaxwellConnection() ) {
+			Long schemaID = findSchema(conn, targetPosition, context.getServerID());
+			if (schemaID == null)
+				return null;
```

Comment:
(hm maybe it isn't possible anymore b/c context now defaults to capturing a new binlog position, which case findSchema would return empty... if not possible to log error, then it's fine)